Cargo.toml

```diff
@@ -41,6 +41,7 @@ kv-log-macro = "1.0.4"
 serde = "1.0.102"
 serde_json = "1.0.41"
 route-recognizer = "0.2.0"
+futures = "0.3.5"
```

Comment:
Does this need to pull in all of `futures`?

Reply:
We can probably only input `futures-util` for this as well.

Reply:
Fwiw `futures-util` has [`FuturesUnordered`](https://docs.rs/futures-util/0.3.5/futures_util/stream/struct.FuturesUnordered.html).

Reply:
addressed in rebase